recipes/emsdk/all/conanfile.py

```diff
@@ -0,0 +1,107 @@
+
+required_conan_version = ">=2.1"
+
+
+class EmSDKConan(ConanFile):
+    name = "emsdk"
+    description = "Emscripten SDK. Emscripten is an Open Source LLVM to JavaScript compiler"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/kripken/emscripten"
+    topics = ("emsdk", "emscripten", "sdk", "emcc", "em++", "nodejs")
+    license = "MIT"
+    package_type = "application"
+    settings = "os", "arch"
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    @property
+    def _emscripten(self):
+        return os.path.join(self.package_folder, "bin", "upstream", "emscripten")
+
+    @property
+    def _node_path(self):
+        subfolders = [path for path in (Path(self.package_folder) / "bin" / "node").iterdir() if path.is_dir()]
+        if len(subfolders) != 1:
+            return None
+        return os.path.join("bin", "node", subfolders[0].name, "bin")
+
+    def generate(self):
+        # To avoid issues when cross-compiling or with not common arch in profiles we need to set EMSDK_ARCH
+        # This is important for the emsdk install command
+        env = VirtualBuildEnv(self)
+        # Special consideration for armv8 as emsdk expects "arm64"
+        arch = "arm64" if str(self.settings.arch) == "armv8" else str(self.settings.arch)
+        env.environment().define("EMSDK_ARCH", arch)
+        env.generate()
+
+    def build(self):
+        with chdir(self, self.source_folder):
+            emsdk = "emsdk.bat" if self.settings_build.os == "Windows" else "./emsdk"
+            self.run(f"{emsdk} install latest")
+            self.run(f"{emsdk} activate latest")
+
+    def package(self):
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        copy(self, "*", src=self.source_folder, dst=os.path.join(self.package_folder, "bin"))
+
+    def finalize(self):
+        copy(self, "*", src=self.immutable_package_folder, dst=self.package_folder)
+        embuilder = os.path.join(
+            self._emscripten, "embuilder" if self.info.settings.os != "Windows" else "embuilder.bat"
+        )
+        self.run(f"{embuilder} build MINIMAL")
+
+    def _define_tool_var(self, value):
+        suffix = ".bat" if self.settings.os == "Windows" else ""
+        path = os.path.join(self._emscripten, f"{value}{suffix}")
+        return path
+
+    def package_info(self):
+        self.cpp_info.bindirs = ["bin", os.path.join("bin", "upstream", "emscripten"), self._node_path]
+        self.cpp_info.includedirs = []
+        self.cpp_info.libdirs = []
+        self.cpp_info.resdirs = []
+
+        # If we are not building for Emscripten, probably we don't want to inject following environment variables,
+        #   but it might be legit use cases... until we find them, let's be conservative.
```

Comment:
The way to be conservative is raise an error. If the recipe is used in some way we are not aware of, for example as a regular requires (without ``settings_target``), then better raise an error, get the user report and learn about the use cases that moving forward and we will never know about it (until it is too late, and they are stuck and we can't unblock)